File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.js
/**
 * @param {number[]} target
 * @param {number} n
 * @return {string[]}
 */
var buildArray = function(target, n) {
    let m = target.length;
    let curr = 1;
    const arr = new Array();

    for(let i = 0; i < m; i++){
        while(curr < target[i]){
            arr.push('Push');
            arr.push('Pop');
            curr++;
        }
        arr.push('Push');
        curr++;
    }
    return arr;
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> str = new ArrayList<>();
        int curr = 1;

        for(int val : target){
            while(curr < val){
                str.add("Push");
                str.add("Pop");
                curr++;
            }
            str.add("Push");
            curr++;
        }
        return str;
    }
}